src/structures/Message.js

```diff
@@ -409,7 +409,9 @@ class Message extends Base {
     return (
       !this.deleted &&
       (this.author.id === this.client.user.id ||
-        (this.guild && this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES, false)))
+        (this.guild &&
+          this.channel.permissionsFor(this.client.user) &&
+          this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES, false)))
```

Comment:
why call this twice?
wouldn't it be better to do something like
```js
const permissions = this.guild && this.channel.permissionsFor(this.client.user);
return !permissions || permissions.has(...);```

Reply:
If #5067 merges first, this line can be refactored to
```suggestion
          this.channel.permissionsFor(this.client.user)?.has(Permissions.FLAGS.MANAGE_MESSAGES, false) ?? false))
```

Reply:
The change @kyranet mentioned can be implemented here now.